src/test/java/org/codehaus/mojo/versions/SetPropertyMojoTest.java

```diff
@@ -39,26 +42,30 @@ public void testNullNewVersion()
             throws Exception
     {
         SetPropertyMojo mojo = createMojo( "set-property",
-                "src/test/resources/org/codehaus/mojo/set-property/null-new-version-pom.xml" );
+                "target/test-classes/org/codehaus/mojo/set-property/pom.xml" );
         assertThat( mojo.getProject().getProperties(), is( mojo.getProject().getModel().getProperties() ) );
-        try
-        {
-            mojo.execute();
-            fail();
-        }
-        catch ( MojoExecutionException e )
-        {
-            assertThat( e.getMessage(),
-                    containsString( "Invalid execution arguments: newVersion must not be empty" ) );
-        }
+
+        setVariableValueToObject( mojo, "property", "dummy-api-version" );
+        setVariableValueToObject( mojo, "newVersion", null );
```

Comment:
What with case when we have empty string? like for `-DnewVersion=`

Reply:
Should be the same, but added a test case just to be safe. Please refresh.